src/app/delegation_backend/src/itn_uptime_analyzer/time.go

```diff
@@ -0,0 +1,121 @@
+	spId := config.AnalyzerOutputGsheetId
+
+	resp, err := client.Spreadsheets.Values.Get(spId, readRange).Do()
+	if err != nil {
+		log.Fatalf("Unable to retrieve data from sheet: %v\n", err)
+	}
+
+	var lastFilledColumn int = len(resp.Values[0]) - 1
+	var lastExecutionBasedOnSheetsAsTime time.Time
+
+	readRange = fmt.Sprintf("%s!%s%d", sheetTitle, string(lastFilledColumn+65), 1)
+
+	lastTimeWindow := resp.Values[0][lastFilledColumn]
+
+	stringSplit := strings.SplitAfter(fmt.Sprint(lastTimeWindow), " - ")
+	lastExecutionBasedOnSheets := stringSplit[len(stringSplit)-1]
+
+	if !strings.HasPrefix(lastExecutionBasedOnSheets, "Node") {
+		lastExecutionBasedOnSheetsAsTime, err = time.Parse(time.RFC3339, lastExecutionBasedOnSheets)
+		if err != nil {
+			log.Fatalf("Unable to parse string to time: %v\n", err)
+		}
+	} else {
+		pastTime := currentTime.Add(-time.Duration(executionInterval) * time.Hour)
+		return pastTime
+	}
+
+	timeDiffHours := time.Since(lastExecutionBasedOnSheetsAsTime).Hours()
+
+	if (lastExecutionBasedOnSheets != "") && (timeDiffHours > time.Since(currentTime.Add(-time.Duration(executionInterval-1)*time.Hour)).Hours()) && (timeDiffHours <= time.Since(currentTime.Add(-time.Duration(executionInterval+1)*time.Hour)).Hours()) {
+		pastTime, err := time.Parse(time.RFC3339, lastExecutionBasedOnSheets)
+		if err != nil {
+			log.Fatalf("Unable to parse string to time: %v\n", err)
+		}
+		return pastTime
+	} else {
+		pastTime := currentTime.Add(-time.Duration(executionInterval) * time.Hour)
+		return pastTime
+	}
+}
+
+// Identifies which sheet should the application write to
+// Currently one sheet should represent one week
+func IdentifyWeek(config AppConfig, client *sheets.Service, log *logging.ZapEventLogger, currentTime time.Time) (string, error) {
+	outputSheets, err := GetSheets(config, client, log)
+
+	if err != nil {
+		log.Fatalf("Error getting sheet names: %v\n", err)
+	}
+
+	lastSheet := outputSheets[len(outputSheets)-1]
+	lastSheetSplit := strings.SplitAfter(fmt.Sprint(lastSheet), " - ")
+	lastSheetEndTime, err := time.Parse(lastSheetSplit[len(lastSheetSplit)-1], time.RFC3339)
+	if err != nil {
+		log.Fatalf("Error parsing time: %v\n", err)
+	}
+
+	readRange := fmt.Sprintf("%s!A%d:Z%d", lastSheet.Properties.Title, 1, 1)
+	spId := config.AnalyzerOutputGsheetId
+
+	resp, err := client.Spreadsheets.Values.Get(spId, readRange).Do()
+	if err != nil {
+		log.Fatalf("Unable to retrieve data from sheet: %v\n", err)
+	}
+
+	var lastFilledColumn int = len(resp.Values[0]) - 1
+
+	currentDate := currentTime.Format("2006-01-02")
+	oneWeekLater := currentTime.Add(7 * 24 * time.Hour).Format("2006-01-02")
+	sheetTitle := strings.Join([]string{currentDate, oneWeekLater}, " - ")
+
+	if lastSheetEndTime.Before(currentTime) {
+		err := CreateSheet(config, client, log, sheetTitle)
+		if err != nil {
+			log.Fatalf("Unable to create new sheet for spreadsheet: %v\n", err)
+		}
+
+		return sheetTitle, nil
+	} else if lastFilledColumn >= 14 {
```

Comment:
This case won't be needed if my suggestion above is implemented. Also, in current implementation it could lead to a confusion when due to some sort of outage analyzer is not launched on a couple of days and then data series will be added to sheet `2023-07-08 - 2023-07-15` for data corresponding to `2023-07-16`